### ISSUE TITLE
[Bug]: CUDA enable_graph=true下运行ais_bench GSM8K 测试失败

### ISSUE THREAD

**Issue by asr-sheep1** [jd-opensource/xllm#1550]
### 1、软件环境
Device: H200
Docker image: quay.io/jd_xllm/xllm-ai:xllm-dev-cuda-x86
Code commit: 71f063df3544da94c39e305b1cc64485787dbc4c
Model: /data/model/Qwen3-8B

### 2、server启动脚本参考quick_start  cuda

#!/bin/bashset -e

export CUDA_VISIBLE_DEVICES=4

MODEL_PATH="/data/model/Qwen3-8B"
MASTER_NODE_ADDR="127.0.0.1:9748"
START_PORT=18001
START_DEVICE=0
LOG_DIR="log"
NNODES=1

mkdir -p $LOG_DIR

for (( i=0; i<$NNODES; i++ ))
do
  PORT=$((START_PORT + i))
  DEVICE=$((START_DEVICE + i))
  LOG_FILE="$LOG_DIR/node_$i.log"
  ./build/xllm/core/server/xllm \
    --model $MODEL_PATH \
    --devices="cuda:$DEVICE" \
    --port $PORT \
    --master_node_addr=$MASTER_NODE_ADDR \
    --nnodes=$NNODES \
    --block_size=32 \
    --max_memory_utilization=0.8 \
    --enable_prefix_cache=false \
    --enable_chunked_prefill=false \
    --enable_schedule_overlap=true \
    --enable_graph=true \
    --max_seqs_per_batch=128 \
    --enable_graph_mode_decode_no_padding=true \
    --node_rank=$i > "$LOG_FILE" 2>&1 &
done

### 3、测试段使用后300条数据可快速复现，配置文件和参数如下:

`tail -n 300 test.jsonl_origin > test.jsonl`
ais_bench/benchmark/configs/models/vllm_api/vllm_api_general_chat.py

from ais_bench.benchmark.models import VLLMCustomAPIChat
from ais_bench.benchmark.utils.model_postprocessors import extract_non_reasoning_content

models = [
    dict(
        attr="service",
        type=VLLMCustomAPIChat,
        abbr="vllm-api-general-chat",
        path="/data/model/Qwen3-8B",
        model="Qwen3-8B",
        request_rate=0,
        retry=2,
        host_ip="localhost",
        host_port=18001,
        max_out_len=32678,
        batch_size=32,
        trust_remote_code=False,
        generation_kwargs=dict(
            temperature=0.6,
            top_p=0.95,
        ),
        pred_postprocessor=dict(type=extract_non_reasoning_content),
    )
]

### 期望行为
开启 CUDA Graph 后，ais_bench GSM8K 测试应能正常完成，不应出现 FlashInfer workspace overflow 或 decode 阶段的 Indexing.cu/c10 ::Error。

### 实际行为【bug】：
1、error1：启动服务时，触发FlashInfer workspace overflow。
     设置 --max_seqs_per_batch=128 后，workspace overflow 可规避该问题，
2、error2: benchmark 运行期间，在处理后半段请求时报错：
`Indexing.cu: indexSelectLargeIndex: Assertion `srcIndex < srcSelectDimSize` failed
terminate called after throwing an instance of 'c10::Error'`

###  AI初步定位与已验证的解决方案
从日志看，max_seqs_per_batch=128 时 decode graph warmup 可以完成，服务也能正常启动。Indexing.cu/c10 ::Error 不是发生在 warmup/capture 阶段，而是发生在实际 ais_bench 请求运行期间。

目前怀疑问题点在 decode 阶段 FlashInfer attention 被捕获进完整 CUDA Graph 后，replay 时使用的 metadata 与实际 batch metadata 不一致。

具体推测如下：

CUDA Graph 开启后，decode 阶段的 FlashInfer attention 会被捕获进完整 CUDA Graph。
decode replay 时，每个 batch 的 paged KV layout、qo_indptr、plan_info 等 metadata 会变化。
虽然 replay 前会更新 persistent metadata，但完整 CUDA Graph replay 不会重新执行 C++ 层的 FlashInfer plan/update 逻辑。
因此 FlashInfer attention replay 可能使用了 capture 阶段的 plan/launch 信息与运行期更新后的 tensor metadata 的不一致组合。
这种不一致可能先导致 attention 输出异常，随后在 logits/sampling 等下游 PyTorch index_select 路径中表现为 indexSelectLargeIndex 越界。
因此，Indexing.cu 更像是下游暴露出来的错误点，而不一定是根因。根因更可能在：

FlashInfer decode attention inside CUDA Graph
特别是 decode graph replay 时 FlashInfer attention metadata / plan_info 的处理。

### 临时验证

Codex 给过一个临时验证方案：参考 prefill 的 piecewise_graph 机制，在 decode 阶段将 FlashInfer attention 从 CUDA Graph 中绕出，只让 attention 前后的计算进入 CUDA Graph。

该方案下，同样的 benchmark 可以完整跑完，未再出现 Indexing.cu/c10 ::Error。

**不过该方案目前仅用于定位和验证，尚不确定是否是合适的最终修复方式，因此暂未提交 PR。**

**Comment by XuZhang99** [comment]
fixed in https://github.com/xLLM-AI/xllm/pull/1939

**Comment by asr-sheep1** [comment]
Tested on my environment. I can no longer reproduce the original issue, and everything works as expected. Thanks for the fix!
